src/mame/sinclair/scorpion.cpp

```diff
@@ -487,6 +563,21 @@ u8 scorpiontb_state::ay_data_r()
 		: m_ay[m_ay_selected]->data_r();
 }
 
+INPUT_CHANGED_MEMBER(scorpiontb_state::turbo_changed)
+{
+	m_turbo = !m_turbo;
+	m_maincpu->set_clock_scale(1 << m_turbo);
+	popmessage("Turbo %s\n", m_turbo ? "ON" : "OFF");
+}
```

Comment:
You should really be checking `newval` here to make sure you don’t get out sync with the state of the toggle input, as it’s saved in the CFG file for the machine (also remember to check it on `machine_reset` to make sure you’re in the right state when starting).

Reply:
Converted to ioport. Per my understanding it should provide proper handling of store state out of the box. Can you take a closer look please?